### ISSUE TITLE
NoSuchPropertyError on higher jackson versions

### ISSUE THREAD

**Issue by JayJayOne** [taboola/api-java-client-core#8]
The `PropertyNamingStrategy.CAMEL_CASE_TO_LOWER_CASE_WITH_UNDERSCORES` on line:
https://github.com/taboola/api-java-client-core/blob/dcd1cf9041f94356c38c4fe799ccb6a938086289/src/main/java/com/taboola/rest/api/internal/serialization/SerializationMapperCreator.java#L12
causes a NoSuchPropertyError on the latest jackson versions. I would've made a PR but due to the fact that all PropertyNamingStrategy items are deprecated i wasn't quite sure how to approach the issue.

Thanks in advance
J.

**Comment by vladimanaev** [comment]
I think we can add a flag similar to what we did with **READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE**
and rename to higher version enum name (I remember it has a new name or so)
```
if(!serializationConfig.shouldDisableReadUnknownEnumValuesAsDefaultValue()) {
...
}
```
Which Jackson version are you using?

**Comment by JayJayOne** [comment]
I've downgraded since, but i think it was 2.16.something. I wanted to upgrade my projects framework and this was part of it. That's why i'm not 100% sure which version it was.

**Comment by vladimanaev** [comment]
@JayJayOne see #9 I am going to add ability to switch to SNAKE_CASE via SerializationConfig flag.
If you still having issues let me know since PropertyNamingStrategy.SNAKE_CASE  is deprecated also from 2.12 and the new way is PropertyNamingStrategies.SNAKE_CASE for which we will need to upgrade  jackson version of SDK.

**Comment by JayJayOne** [comment]
@vladimanaev Sorry for taking some time to respond i just got back working on that project. Could you update the dependency in the backstage-api-java-client? I'm actually using this library and since there hasn't been any update yet the problem still persists. 
Thanks J.

**Comment by vladimanaev** [comment]
@JayJayOne released as part of 1.1.8 of Backstage repository
